packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp

```diff
@@ -73,20 +73,39 @@ namespace DefaultTypes {
 
   /// \typedef global_ordinal_type
   /// \brief Default value of GlobalOrdinal template parameter.
-#if defined(HAVE_TPETRA_INST_INT_INT)
-  typedef int global_ordinal_type;
-#elif defined(HAVE_TPETRA_INST_INT_LONG_LONG)
-  typedef long long global_ordinal_type;
-#elif defined(HAVE_TPETRA_INST_INT_LONG)
-  typedef long global_ordinal_type;
-#elif defined(HAVE_TPETRA_INST_INT_UNSIGNED_LONG)
-  typedef unsigned long global_ordinal_type;
-#elif defined(HAVE_TPETRA_INST_INT_UNSIGNED)
-  typedef unsigned global_ordinal_type;
-#else
-#  error "Tpetra: No global ordinal types in the set {int, long long, long, unsigned long, unsigned} have been enabled."
+#if defined(TPETRA_ENABLE_DEPRECATED_CODE)
+    // NEW style picks `int` for Global Ordinal as the first choice.
+    #if defined(HAVE_TPETRA_INST_INT_INT)
+        typedef int global_ordinal_type;
+    #elif defined(HAVE_TPETRA_INST_INT_LONG_LONG)
+      typedef long long global_ordinal_type;
+    #elif defined(HAVE_TPETRA_INST_INT_LONG)
+      typedef long global_ordinal_type;
+    #elif defined(HAVE_TPETRA_INST_INT_UNSIGNED_LONG)
+      typedef unsigned long global_ordinal_type;
+    #elif defined(HAVE_TPETRA_INST_INT_UNSIGNED)
+      typedef unsigned global_ordinal_type;
+    #else
+        #error "Tpetra: No global ordinal types in the set {int, long long, long, unsigned long, unsigned} have been enabled."
+    #endif
+#else  // TPETRA_ENABLE_DEPRECATED_CODE IS NOT DEFINED
+    // NEW style picks `long long` for Global Ordinal as the first choice.
```

Comment:
The order doesn't actually matter here, since exactly one of these macros is allowed to be defined.

Reply:
True with the changes to CMake that enforce only 1 allowable GORD to be set, but reordering that switch statement was specifically requested by Karen and the notes from our 4/9 meeting saying to modify the header to "pick long long" as the default was interpreted to mean putting long long first in the if/elseif/else chain.

Reply:
@william76 It's cool, I don't mind :)